src/main/java/org/sonatype/goodies/packageurl/PercentEncoding.java

```diff
@@ -70,15 +80,41 @@ public static String encodeQualifierValue(final String value) {
     return encode(value);
   }
 
-  // NOTE: guava doesn't provide any decoding support, but URLDecoder should properly decode value
-
   public static String decode(final String value) {
-    checkNotNull(value);
+    requireNonNull(value);
     try {
       return URLDecoder.decode(value, UTF_8);
     }
     catch (UnsupportedEncodingException e) {
       throw new RuntimeException(e);
     }
   }
+
+  /**
+   * A simpler version of String.replace() that does not involve regexp's and Patterns.
+   * Modeled after a similar method in org.codehaus.plexus's StrungUtils.
+   * https://github.com/codehaus-plexus/plexus-utils/blob/master/src/main/java/org/codehaus/plexus/util/StringUtils.java#L848
+
+   */
+  // @VisibleForTesting
+  static String simpleReplace(String text, String target, String replacement) {
```

Comment:
could this be replaced with the JDK's `String#replace(CharSequence, CharSequence)`?

Reply:
The reason is in the comment:
```
A simpler version of String.replace() that does not involve regexp's and Pattern
```

I started out with `String.replace()` but in Jason's review, he objected to the performance hit...

Reply:
@bradcupit as @andrewstein says, String.replace() will actually Pattern.compile() on each invocation; its leaning on the regex muck to do what is otherwise a simple string manipulation operation.  It may, as I didn't dig too deep into the JDK's impl of Pattern.LITERAL usage in a compiled pattern short-circuit some of the regex-foo, but its still diving pretty deep into regex-land.

I originally suggested pre-compiling these patterns, but then figured it may just be simpler and to lift a simple string replacer method from a public utility collection like commons-lang or plexus-utils instead.

Reply:
nice, I learned something new today. I knew the other replace* methods used regex's but didn't realize `replace(CharSequence, CharSequence)` used one behind the scenes.

suggestion withdrawn